### PR TITLE
Handle renaming column casing

### PR DESCRIPTION
**Pull request by dpsutton**
fixes https://github.com/metabase/metabase/issues/7923

products -> Products and the like

need to worry about one issue though:
what if they _add_ a new column that has a name that is the same but
for casing:
products and Products. We would probably identify arbitrarily one of
these with the existing products column and probably bad things will
happen. Need to extend this check that identifies the casing
indifference only in the case that there is no ambiguity

### CODE REVIEW

File: test/metabase/sync/sync_metadata/fields_test.clj [hunk 1]
```diff
@@ -47,127 +49,146 @@
...
+    (let [db-state (with-test-db-before-and-after-altering
+                     "ALTER TABLE \"birds\" RENAME COLUMN \"example_name\" to \"Example_Name\";"
+                     (fn [database]
+                       (set
+                        (map (partial into {})
+                             (db/select [Field :id :name :active]
+                                        :table_id [:in (db/select-ids Table :db_id (u/get-id database))])))))]
+      (is (= {:before-drop #{{:name "species",      :active true}
+                             {:name "example_name", :active true}}
+              :after-drop #{{:name "species",      :active true}
```
Comment (tsmacdonald) [inline test/metabase/sync/sync_metadata/fields_test.clj:70]: Minor: Maybe `:before-sync` and `:after-sync` are better names?

**Review by tsmacdonald — APPROVED** [review]

File: test/metabase/sync/sync_metadata/fields_test.clj (hunk 1 above)
Comment (dpsutton) [reply test/metabase/sync/sync_metadata/fields_test.clj]: good catch. done!

**Comment by dpsutton** [comment]
rebased on fixed release-38 timezone tests. waiting on green

**Review by camsaul — APPROVED** [review]
Cool

**Pull request merged**